### PR TITLE
[#47071] Remove budget activity items from the activity list

### PR DESCRIPTION
**Pull request by mrmir**
https://community.openproject.org/work_packages/47071

### CODE REVIEW

File: app/models/activities/fetcher.rb [hunk 1]
```diff
@@ -57,6 +57,7 @@ def event_types
         else
           OpenProject::Activity.available_event_types.to_a
         end
+      .excluding('budgets')
```
Comment (ulferts) [inline app/models/activities/fetcher.rb]: The cleaner way is to remove `Activities::BudgetActivityProvider` and the registration of it in the budget's engine.

I thought about also removing the journals while we are at it on account of them not carrying a lot of valuable information but that might be too hasty.

**Review by ulferts — CHANGES_REQUESTED** [review]
Hey @mrmir, please see my feedback inline.

**Comment by mrmir** [comment]
Hey @ulferts, I have pushed the changes to remove `Activities::BudgetActivityProvider` and its registration in the budget engine. 

Is it okay now?

**Comment by mrmir** [comment]
Hey @ulferts, I have pushed the changes to remove `Activities::BudgetActivityProvider` and its registration in the budget engine. 

Is it okay now?

File: spec/models/activities/fetcher_integration_spec.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@
 
 describe Activities::Fetcher, 'integration' do
   shared_let(:user) { create(:user) }
-  shared_let(:permissions) { %i[view_work_packages view_time_entries view_changesets view_wiki_edits] }
+  shared_let(:permissions) { %i[view_work_packages view_time_entries view_changesets view_wiki_edits view_budgets] }
```
Comment (ulferts) [inline spec/models/activities/fetcher_integration_spec.rb]: With the `BudgetActivityProvider` removed, I don't think the changes in this file are necessary any more. `budgets` can no longer be part of the activities to fetch. 

With such a change it sometimes makes sense to test for the removed functionality to no longer have an effect if a user by some means triggers the functionality (e.g by having a URL stored as a bookmark). 

With that in mind, I'd test that `Activities::Fetcher.new(some_user).event_types` does no longer include `'budgets'`.

**Review by ulferts — CHANGES_REQUESTED** [review]
Almost there @mrmir

File: spec/models/activities/fetcher_integration_spec.rb (hunk 2 above)
Comment (mrmir) [inline spec/models/activities/fetcher_integration_spec.rb]: @ulferts I have removed all the extra tests and added what I believe you are asking for. Not sure if the `view_budgets` permission needs to also be removed, however.

Comment (ulferts) [reply spec/models/activities/fetcher_integration_spec.rb]: Hi @mrmir simple way to test that: If you now revert the removal of the `BudgetActivityProvider` and run the test, will it become red? And is it still red if the `view_budgets` permission is removed?

Comment (mrmir) [reply spec/models/activities/fetcher_integration_spec.rb]: Aah thanks @ulferts. As the test stayed red without the permission as well, I have removed it.

**Review by ulferts — APPROVED** [review]
👍

**Pull request merged**